design/src/docs/anatomy.md

```diff
@@ -0,0 +1,687 @@
+
+The `from_handler` constructor is used in the following way:
+
+```rust
+async fn get_pokemon_service(input: GetPokemonServiceInput) -> Result<GetPokemonServiceOutput, GetPokemonServiceError> {
+    /* Handler logic */
+}
+
+let operation = GetPokemonService::from_handler(get_pokemon_service);
+```
+
+Alternatively, `from_service` constructor:
+
+```rust
+struct Svc {
+    /* ... */
+}
+
+impl Service<GetPokemonServiceInput> for Svc {
+    type Response = GetPokemonServiceOutput;
+    type Error = GetPokemonServiceError;
+
+    /* ... */
+}
+
+let svc: Svc = /* ... */;
+let operation = GetPokemonService::from_service(svc);
+```
+
+To summarize, the `S`, in `Operation<S, L>`, is a _model service_ constructed from a `Handler` or a `OperationService` subject to the constraints of an `OperationShape`.
+
+Now, what about the `L` in `Operation<S, L>`? The `L` is a `tower::Layer`, or colloquially "middleware", that is applied to a _HTTP service_. Note that this means that `L` is _not_ applied directly to `S`. We can append to `L` using the `Operation::layer` method:
+
+```rust
+impl<S, L> Operation<S, L> {
+    /// Applies a [`Layer`] to the operation _after_ it has been upgraded via [`Operation::upgrade`].
+    pub fn layer<NewL>(self, layer: NewL) -> Operation<S, Stack<L, NewL>> {
+        Operation {
+            inner: self.inner,
+            layer: Stack::new(self.layer, layer),
+        }
+    }
+}
+```
+
+where [`Stack`](https://docs.rs/tower/latest/tower/layer/util/struct.Stack.html) is used to chains layers together.
+
+A typical use of this might be:
+
+```rust
+let operation = GetPokemonSpecies::from_handler(handler).layer(RequestBodyLimitLayer::new(500));
+```
+
+where [`RequestBodyLimitLayer`](https://docs.rs/tower-http/latest/tower_http/limit/struct.RequestBodyLimitLayer.html) limits the size of the HTTP request to the `GetPokemonSpecies` operation.
+
+As mentioned, `L` is applied _after_ the `Operation<S, L>` has been "upgraded" to a HTTP service. The procedure of upgrading a model service to a HTTP service is described in the [Upgrading a Model Service](#upgrading-a-model-service) section below.
+
+## Serialization and Deserialization
+
+A [Smithy protocol](https://awslabs.github.io/smithy/2.0/spec/protocol-traits.html#serialization-and-protocol-traits) specifies the serialization/deserialization scheme - how a HTTP request is transformed into a modelled input and a modelled output to a HTTP response. The is formalized using the [`FromRequest`](https://github.com/awslabs/smithy-rs/blob/4c5cbc39384f0d949d7693eb87b5853fe72629cd/rust-runtime/aws-smithy-http-server/src/request.rs#L156-L164) and [`IntoResponse`](https://github.com/awslabs/smithy-rs/blob/4c5cbc39384f0d949d7693eb87b5853fe72629cd/rust-runtime/aws-smithy-http-server/src/response.rs#L40-L44) traits:
+
+```rust
+/// Provides a protocol aware extraction from a [`Request`]. This consumes the
+/// [`Request`], in contrast to [`FromParts`].
+pub trait FromRequest<Protocol>: Sized {
```

Comment:
These traits, and others all have a loose counterpart in `axum` e.g. https://docs.rs/axum/latest/axum/extract/trait.FromRequest.html because

1. We used to used `axum` extensively
2. These traits are reasonably natural and occur across all frameworks ([Rocket](https://api.rocket.rs/v0.4/rocket/request/trait.FromRequest.html, [Actix](https://docs.rs/actix-web/latest/actix_web/trait.FromRequest.html), )

Should we mention this?

Reply:
I don't think it's necessary, it's a fairly common approach in the Rust web ecosystem (as you stated).